### DISCUSSION TITLE
API Improvement

### DISCUSSION THREAD

**Discussion by H3rnand3zzz** [profanity-im/profanity#1928] [Ideas]
Current API can benefit from the following features:
 - Redraw UI (needed if plugin invokes things like `/editor` by itself)
 - Access to profanity settings (e.g. `/privacy` settings)
 - Profanity's location (to access DB, configs, scripts and other things that API don't provide direct access to)
 - `pre_command`, `post_command` to interact better with commands
 - More interaction with readline (typing bar) to add/replace/read input, for example: replace -- (double en dash) to — (em dash), replace emoticons to unicode etc.
 - (potentially) add key handling to allow keybindings, also allow access to some internal profanity features like scrolling

This list is a subject to change.

**Comment by jubalh** [comment]
The API was neglected a long time. I'm happy you want to improve it.
Depending on how it goes I would set this as a goal for 0.16 and not 0.15 but you certainly can work on it.
If a whole API overhaul is needed that is fine too if we increase the number.
